modoboa/lib/email_utils.py

```diff
@@ -262,6 +263,47 @@ def render_headers(self, **kwargs):
         return render_to_string("common/mailheaders.html", context)
 
 
+def split_address(address):
+    """Split an e-mail address into local part and domain."""
+    assert isinstance(address, six.text_type),\
+        "address should be of type %s" % six.text_type.__name__
+    if "@" not in address:
+        local_part = address
+        domain = None
+    else:
+        local_part, domain = address.split("@", 1)
+    return (local_part, domain)
+
+
+def split_local_part(local_part, delimiter=None):
+    """Split a local part into local part and extension."""
+    assert isinstance(local_part, six.text_type),\
+        "local_part should be of type %s" % six.text_type.__name__
+    assert isinstance(delimiter, six.text_type) or delimiter is None,\
+        "delimiter should be of type %s" % six.text_type.__name__
+    extension = None
+    if local_part.lower() in ["mailer-daemon", "double-bounce"]:
```

Comment:
These entries should be set as settings, because there may be custom addresses ?
eg:
LOCAL_PART_IGNORE = ["mailer-daemon", "double-bounce"]

Reply:
Good suggestion, however mailer-daemon, double-bounce, owner-* and *-request are the only special addresses that postfix and amavis deal with.

See http://www.postfix.org/postconf.5.html#recipient_delimiter

Reply:
I think we can start like this and make it configurable if we receive requests.